src/cli.rs

```diff
@@ -0,0 +1,764 @@
+    /// '--change-newer-than' or '--newer' can be used as aliases.
+    /// Examples:
+    ///     --changed-within 2weeks
+    ///     --change-newer-than '2018-10-27 10:00:00'
+    ///     --newer 2018-10-27
+    #[clap(
+        long,
+        alias("change-newer-than"),
+        alias("newer"),
+        value_name = "date|dur",
+        number_of_values = 1,
+        verbatim_doc_comment,
+        action
+    )]
+    pub changed_within: Option<String>,
+    /// Filter by file modification time (older than)
+    ///
+    /// The argument can be provided as a specific point in time (YYYY-MM-DD HH:MM:SS)
+    /// or as a duration (10h, 1d, 35min).
+    /// '--change-older-than' or '--older' can be used as aliases.
+    ///
+    /// Examples:
+    ///     --changed-before '2018-10-27 10:00:00'
+    ///     --change-older-than 2weeks
+    ///     --older 2018-10-27
+    #[clap(
+        long,
+        alias("change-older-than"),
+        alias("older"),
+        value_name = "date|dur",
+        number_of_values = 1,
+        verbatim_doc_comment,
+        action
+    )]
+    pub changed_before: Option<String>,
+    /// Limit number of search results
+    ///
+    /// Limit the number of search results to 'count' and quit immediately.
+    #[clap(long, value_name = "count", hide_short_help = true, value_parser)]
+    max_results: Option<usize>,
+    /// Limit search to a single result and quit immediately
+    ///
+    /// This is an alias for '--max-results=1'.
+    #[clap(
+        short = '1',
+        hide_short_help = true,
+        overrides_with("max-results"),
+        action
+    )]
+    max_one_result: bool,
+    /// Print nothing, exit code 0 if match found, 1 otherwise
+    ///
+    /// When the flag is present, the program does not print anything and will
+    /// return with an exit code of 0 if there is at least one match. Otherwise, the
+    /// exit code will be 1.
+    ///
+    /// '--has-results' can be used as an alias.
+    #[clap(
+        long,
+        short = 'q',
+        alias = "has-results",
+        hide_short_help = true,
+        conflicts_with("max-results"),
+        action
+    )]
+    pub quiet: bool,
+    /// Show filesystem errors
+    ///
+    ///Enable the display of filesystem errors for situations such as
+    ///insufficient permissions or dead symlinks.
+    #[clap(long, hide_short_help = true, overrides_with("show-errors"), action)]
+    pub show_errors: bool,
+    /// Change current working directory
+    ///
+    /// Change the current working directory of fd to the provided path. This
+    /// means that search results will be shown with respect to the given base
+    /// path. Note that relative paths which are passed to fd via the positional
+    /// <path> argument or the '--search-path' option will also be resolved
+    /// relative to this directory.
+    #[clap(
+        long,
+        value_name = "path",
+        number_of_values = 1,
```

Comment:
Why is `number_of_values` set?

Reply:
It was set in the old code. but I suppose with `action` specified it is probably redundant?

Reply:
Its also one of those things that a builder takes care of for you